typescript/cli/src/deploy/core.ts

```diff
@@ -109,8 +111,9 @@ export async function runCoreDeploy(params: DeployParams) {
 
       const coreModule = await AltVMCoreModule.create({
         chain,
-        config,
-        multiProvider,
+        // TODO: Remove this cast when all ISM and Hook types are supported
+        config: config as ProviderCoreConfig,
```

Comment:
I see some of these will be fixed by https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/7423 so probably worth having another look after it is merged